docs/design/solutions/path-validation/path-validation.md

```diff
@@ -0,0 +1,80 @@
+# Path Validation
+
+## Motivation
+The main use case is to allow users to validate if some arbitrary flow path is possible to create with the given 
+constraints without actually creating this flow. A path here is a sequence of nodes, that is a sequence of switches
+with in and out ports. Constraints are various parameters such as max latency, bandwidth, encapsulation type, 
+path computation strategy, and other.
+
+## Implementation proposal
+Reuse PCE components (specifically extend AvailableNetwork) for the path validation in the same manner as it is used for
```

Comment:
I think you're going to extend not only AvailableNetwork but also the finders. I mean, the only way to check that a path accomplish max_latency, or cost_and_available_bandwidth is to find the actual path, and compare it with the given one. And the most of times it will be only one available path.

BTW, take also in care that the AvailableNetwork (right now) is going to give you the network isls that already have enough bandwidth, so you are not going to be able to see the complete network.

Reply:
I didn't know that about bandwidth, thanks! From one perspective, available bandwidth is one of the parameters, if there's not enough bandwidth, this path is invalid. But it then brings the question whether this is a mandatory parameter. Also, in case of not sufficient bandwidth, there will be no way to validate other parameters via AvailableNetwork.

Reply:
max_bandwidth is not mandatory. At least user always can set it to 0